data/src/main/java/com/alexandruvalentinconstantin/data/NewsRepository.java

```diff
@@ -0,0 +1,29 @@
+package com.alexandruvalentinconstantin.data;
+
+import com.alexandruvalentinconstantin.data.feature.news.local.ArticleEntity;
+import com.alexandruvalentinconstantin.data.feature.news.model.Article;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.annotations.NonNull;
+
+public interface NewsRepository {
+
+    @NonNull
+    Single<List<Article>> getNewsArticles();
+
+    @NonNull
+    Single<List<Article>> getNewsList();
+
+    @NonNull
+    Single<ArticleEntity> getNewsItem(int itemId);
+
+    @NonNull
+    Completable saveArticleItem(ArticleEntity toDos);
+
+    @NonNull
+    Completable deleteItem(int itemId);
```

Comment:
we could reduce this interface, exposing the repository for saveArticleItem and delete item, can bring less encapsulation, might allow the UI to break. Considering deleteItem and saveArticleItem are not needed, let's delete them